pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,506 @@
+
+            self._coeffs = (
+                simplified_coeffs  # Losing gradient in case of torch interface at this point
+            )
+
+            self._ops = simplified_ops
+            with qml.QueuingManager().stop_recording():
+                operands = [qml.s_prod(c, op) for c, op in zip(self._coeffs, self._ops)]
+
+            super().__init__(*operands, id=id, _pauli_rep=pr)
+
+    def _check_batching(self):
+        """Override for LinearCombination, batching is not yet supported."""
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        decimals = None if (len(self.parameters) > 3) else decimals
+        return super(CompositeOp, self).label(
+            decimals=decimals, base_label=base_label or "𝓗", cache=cache
+        )  # Skipping the label method of CompositeOp # pylint: disable=bad-super-call
+
+    @property
+    def coeffs(self):
+        """Return the coefficients defining the LinearCombination.
+
+        Returns:
+            Iterable[float]): coefficients in the LinearCombination expression
+        """
+        return self._coeffs
+
+    @property
+    def ops(self):
+        """Return the operators defining the LinearCombination.
+
+        Returns:
+            Iterable[Observable]): observables in the LinearCombination expression
+        """
+        return self._ops
+
+    def terms(self):
+        r"""TODO"""
+        return self.coeffs, self.ops
+
+    @property
+    def wires(self):
+        r"""The sorted union of wires from all operators.
+
+        Returns:
+            (Wires): Combined wires present in all terms, sorted.
+        """
+        return self._wires
+
+    @property
+    def name(self):
+        return "LinearCombination"
+
+    @qml.QueuingManager.stop_recording()
+    def _simplify_coeffs_ops(self, cutoff=1.0e-12):
+        """Simplify coeffs and ops
+
+        Returns:
+            coeffs, ops, pauli_rep"""
+
+        if len(self.ops) == 0:
+            return [], [], self.pauli_rep
+
+        # try using pauli_rep:
+        if pr := self.pauli_rep:
+
+            wire_order = self.wires
+            if len(pr) == 0:
+                return [], [], pr
+
+            # collect coefficients and ops
+            coeffs = []
+            ops = []
+
+            for pw, coeff in pr.items():
+                pw_op = pw.operation(wire_order=wire_order)
+                ops.append(pw_op)
+                coeffs.append(coeff)
+
+            return coeffs, ops, pr
+
+        if len(self.ops) == 1:
+            return self.coeffs, [self.ops[0].simplify()], pr
+
+        op_as_sum = qml.sum(*self.operands)
+        op_as_sum = op_as_sum.simplify(cutoff)
+        coeffs, ops = op_as_sum.terms()
+        return coeffs, ops, None
+
+    def simplify(self, cutoff=1.0e-12):
+        coeffs, ops, pr = self._simplify_coeffs_ops(cutoff)
+        return LinearCombination(coeffs, ops, _pauli_rep=pr)
+
+    def _obs_data(self):
```

Comment:
I would also consider this a deprecated logic that we no longer want to rely on.

Reply:
yep 💯 but right now we still need it for feature parity with `Hamiltonian` right?

Reply:
I would call this more an unfortunate choice of implementation details.